Source/Renderer/PickFramebuffer.js

```diff
@@ -114,6 +114,9 @@ define([
 
                 var object = context.getObjectByPickColor(colorScratch);
                 if (defined(object)) {
+                    if (object.id && (typeof object.id === 'object')) {
+                        return (object.id.selectable) ? object : undefined;
```

Comment:
@bradAnderson58 the problem here is that `id` is not always guaranteed to be an `Entity` instance, so we can't rely on this check here.

We don't have a top-level Entity picking API right now, so this would have to be done in user code.  We want to add one though, `DataSourceDisplay` would probably be a good place for that.

I think the bigger problem is that this can't be implemented efficiently without such functionality at the primitive level.  We basically would have to have every primitive have the `selectable` property, then keep a similar check here.  However, I don't think `selectable` is the right approach for all use cases.  I would rather see a `pickPriority` property as discussed in #1592.  It would be a number where `<=0` means do not pick and higher numbers means "pick this first".  This has drill pick ramifications but shouldn't be too hard optimize for most use cases.  If we added this feature at the primitive level, we would then have to thread it through up to the Entity API and visualizers.

If @pjcozzi is on board with changing the Primitive API and you are interested in working on this, I can write up some more details instructions about what I think has to happen.

Also, sorry for taking so long to look at this PR.